tests/data/json/profile_with_alter_subparts.json

```diff
@@ -0,0 +1,76 @@
+{
+  "profile": {
+    "uuid": "A0000000-0000-4000-8000-000000000000",
+    "metadata": {
+      "title": "Profile with subparts",
+      "last-modified": "2021-01-01T00:00:00.000+00:00",
+      "version": "2021-01-01",
+      "oscal-version": "1.0.0"
+    },
+    "imports": [
+      {
+        "href": "trestle://catalogs/nist_cat/catalog.json",
+        "include-controls": [
+          {
+            "with-ids": [
+              "ac-1",
+              "ac-2"
+            ]
+          }
+        ]
+      }
+    ],
+    "merge": {
+      "as-is": true
+    },
+    "modify": {
+      "alters": [
+        {
+          "control-id": "ac-1",
+          "adds": [
+            {
+              "position": "after",
+              "by-id": "ac-1_smt",
+              "parts": [
+                {
+                  "id": "ac-1_implgdn",
+                  "name": "ImplGuidance",
+                  "prose": "New part in subpart"
+                },
+                {
+                  "id": "ac-1_expected_evidence",
+                  "name": "expected_evidence",
+                  "prose": "Some evidence prose."
+                },
+                {
+                  "id": "ac-1_above_the_line_guidance",
+                  "name": "above_the_line_guidance",
+                  "prose": "",
+                  "parts": [
+                    {
+                      "id": "ac-1_above_the_line_guidance.part_a",
+                      "name": "part_a.",
+                      "prose": "",
+                      "parts": [
```

Comment:
This isn't quite right because the one statement part in a control is special and its parts are all named "item".  The label of a part, such as "a." is a property with name "label" and value "a."  If you are looking for things by name in the statement, you will need to look for the associated label property.

New guidance added as a subpart of "a." is not currently given a label or title or property and the corresponding section is only identifiable through the last part of the id after the final ".".  

I think we should go ahead and add a property with a label and default to using the id if it is not found.  That way your code can access by label - but note that the label is not unique in the statement - e.g. there are multiple "1."

https://pages.nist.gov/OSCAL/concepts/layer/control/catalog/sp800-53rev5-example/ has some discussion of this issue but no example of added guidance to a statement part.  I may need to ask on the NIST gitter if added guidance should have name "item" or not.

When guidance is added as a subpart of a statement, its name will be "item" - but when guidance is added as a subpart of some other part (not the one statement) it is given a descriptive name.

If you look at control_subparts_dict in profile_test.py it spells out these different cases and the values for the part.id, part.name and part.prose.

So to avoid confusion I would map this profile to the way ac-1 is actually laid out in the nist catalog - and use "a." rather than part_a.

See _add_sub_parts() for how the id_ is created - and note that the id may be overwritten by the fixed_part_name if provided.